spec/controllers/books_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+require './spec/shared_context.rb'
+
+RSpec.describe "Books", type: :controller do
+  describe 'GET #index' do
+    context 'When fetching all the books' do
+      let!(:books) { create_list(:book, 5) }
+
+      it 'responses with the books json' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(
+          books, each_serializer: BookSerializer
+        ).to_json
+        expect(response.body.to_json) =~ JSON.parse(expected)
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe 'GET #show' do
+    context 'When fetching a book' do
```

Comment:
Here you handle the ep responses so you should think what could happens when someone consumes this ep. For example you can do more contexts to cover more possibilities.  For example when the books exists or when the book don't exists.